.travis.yml

```diff
@@ -2,16 +2,26 @@ language: python
 sudo: false
 python:
     - 2.7
-    - 3.3
     - 3.4
     - 3.5
+    - 3.6
     - pypy
-#   PyPy3 support pending a release of a fix for
-#   https://bitbucket.org/pypy/pypy/issue/1946)
-#   - pypy3
+    - pypy3
+matrix:
+    include:
+        - python: "3.7"
+          dist: xenial
+          sudo: true
 install:
-    - pip install .
+    - pip install -U pip setuptools
+    - pip install -U coverage coveralls
+    - pip install -U -e .[test]
 script:
-    - python setup.py -q test -q
+    - COVERAGE_HOME=. COVERAGE_PROCESS_START=./.coveragerc coverage run -m zope.testrunner --test-path=src
```

Comment:
I suspect the subprocesses may be changing the current working directory, so try setting `COVERAGE_HOME=$PWD` or `COVERAGE_HOME=$(pwd)`, and `COVERAGE_PROCESS_START=COVERAGE_HOME/.coveragerc`.

(Actually believe Travis concatenates all the commands into one shell script it executes, so you could put those on a separate line above and `export` them, to avoid unreadable overlong lines.)

Reply:
Let's give it a try.